### DISCUSSION TITLE
Cannot get text lines for this pdf, what is reason?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2075] [Looking for help] [answered]
[cnjpTest.pdf](https://github.com/pymupdf/PyMuPDF/files/10056734/cnjpTest.pdf)
import fitz

path = r'F:\cnjpTest.pdf'

doc = fitz.open(path)

j = 0
for i in range(0, doc.page_count):
    page = doc[i]
    for block in page.get_text("dict", flags=fitz.TEXTFLAGS_TEXT)["blocks"]:
        for line in block["lines"]:
            j = j + 1
            bbox = line["bbox"]  # the line bbox
            text = " ".join([span["text"] for span in line["spans"]])  # text in line
            if len(text.strip()) > 2:
                print(bbox)
                print(text)
                pix = page.get_pixmap(clip=bbox, dpi=300)  # pixmap of line bbox
                # pix.save(...)"page%s-%s.png" % (page, xref)
                pix.save(r'F:\0mupdflines\\' + "page-%i_%i.png" % (page.number, j))  # store image as a PNG
    # doc.save(path+'222.pdf')

doc.close()

**Comment by JorjMcKie** [comment, marked as answer]
Contains no text!

**Reply by nissansz** [reply]
Any command to save docx as pdf for pymupdf?

Above sample is saved from MS word as pdf directly.
I will try to print with acrobat.

**Reply by JorjMcKie** [reply]
> Any command to save docx as pdf for pymupdf?

No, of course not! PyMuPDF reads the same PDF variety as any other tool.

>  I will try to print with acrobat.

You don't believe me? Well, good luck.

**Comment by nissansz** [comment]
Why the text is missing when I converted from docx to pdf? All become non-text elements.

**Reply by JorjMcKie** [reply]
I don't know your Word document. Maybe it was just images.
The page has hundreds of single images.

**Comment by nissansz** [comment]
[cnjpTest.docx](https://github.com/pymupdf/PyMuPDF/files/10056956/cnjpTest.docx)
It is editable in docx. I used effect for font.

**Reply by JorjMcKie** [reply]
Aha, so Word has decided it needs to use images to achieve a similar visual impression.

**Reply by JorjMcKie** [reply]
LibreOffice - as opposed to Word - often is better when exporting to PDF.
You want to try this alternative.

**Reply by Augus1999** [reply]
> [cnjpTest.docx](https://github.com/pymupdf/PyMuPDF/files/10056956/cnjpTest.docx) It is editable in docx. I used effect for font.

It looks you used a text style that cannot be rendered correctly in pdf so MS Word decided to convert them into images. XD

**Comment by nissansz** [comment]
![image](https://user-images.githubusercontent.com/41010669/203076272-3a73e87d-2156-443a-8af4-bc64fa4c780a.png)
Background is missing when open the same docx file, don't know how to insert background image

**Reply by JorjMcKie** [reply]
As always (I told you once): `page.insert_image(rect,...., overlay=False)`.

**Reply by JorjMcKie** [reply]
Similar is possible in the Word document - forgot where to look that one up in the Word menu.
